repo/migrations/Migration012_test.go

```diff
@@ -0,0 +1,564 @@
+package migrations_test
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"regexp"
+	"testing"
+
+	// crypto "gx/ipfs/QmaPbCnUMBohSGo3KnxEa2bHqyJVVeEEcwtqJAYxerieBo/go-libp2p-crypto"
+	"github.com/OpenBazaar/jsonpb"
+	"github.com/OpenBazaar/openbazaar-go/pb"
+	"github.com/OpenBazaar/openbazaar-go/repo/migrations"
+	"github.com/gogo/protobuf/proto"
+	"github.com/ipfs/go-ipfs/repo/fsrepo"
+	"github.com/libp2p/go-libp2p-crypto"
+	peer "github.com/libp2p/go-libp2p-peer"
+)
+
+const testMigration012_IdentityKeyBase64 = "CAESYHwrVuRp5s2u0w5ykibsR77aHWBmvpcaDq+vU9pv8lOqae31NJYJbdDsOlxVRqQZS/eDfssdd7N/rJmoVbQvPytp7fU0lglt0Ow6XFVGpBlL94N+yx13s3+smahVtC8/Kw=="
+
+var (
+	testMigration012_datadir          = path.Join("/", "tmp", "openbazaar-test")
```

Comment:
Can we build a new repo for each test with a random path? `github.com/OpenBazaar/openbazaar-go/schema.GenerateTempPath()`. The fixed directory lends itself to leakage between test runs.